src/main/java/duke/DukeController.java

```diff
@@ -0,0 +1,245 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class DukeController {
```

Comment:
[Code Quality] `DukeController` as a class is probably too broad. Try to figure out how to enforce SRP and better separation of concerns. Also, please add a comment for the class if it's not trivial.